### PR TITLE
Add promo src query param to install urls

### LINKED ISSUE #966: Add src to the addon url used by install
We need to add `src=discovery-promo` to the add-on urls used by the install so that they're picked up in AMO stats.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #966

### CODE REVIEW

File: src/core/utils.js [hunk 1]
```diff
@@ -115,3 +115,11 @@ export function isAllowedOrigin(urlString, { allowedOrigins = [config.get('amoCD
   }
   return allowedOrigins.includes(`${parsedURL.protocol}//${parsedURL.host}`);
 }
+
+export function addQueryParams(urlString, queryParams = {}) {
+  const urlObj = url.parse(urlString, true);
+  // Clear search so we always use query.
+  urlObj.search = null;
```
Comment (mstriemer) [inline src/core/utils.js]: Why do you need to clear this?

File: tests/client/core/test_utils.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
+import url from 'url';
+
```
Comment (mstriemer) [inline tests/client/core/test_utils.js]: I'm surprised this line passes the linter. I don't think it should be there?

**Comment by mstriemer** [comment]
The helper is pretty simple so it feels like it's just duplicating the tests in `url` but that's probably fine.

r+wc

File: src/core/utils.js (hunk 1 above)
Comment (muffinresearch) [inline src/core/utils.js]: Because query is only used if search doesn't exist.

File: tests/client/core/test_utils.js (hunk 2 above)
Comment (muffinresearch) [inline tests/client/core/test_utils.js]: It's because this lib is the same name as a stdlib module so the linter expects it to be in a group on its own.

**Pull request merged**